### PR TITLE
Add new resource

### PR DESCRIPTION
**Pull request by nunibaranes**
- Adding 2 options to ActionMenu
    - Create Article
    - Create Video
    (change skip property to false for seeing in the UI - I'm hiding it until the feature will be ready)
- Clicks on one of the options should appear in the outline new item and the details panel should open, currently, this part is missing in the SDK so appears lesson instead of an item. 
- Created 2 form components, one for a new base resource, and for video.
- Changed ActionMenu button styles
- Adding an optional `tooltipTitle` prop to AddButton.

### CODE REVIEW

File: src/components/Screen/CourseEditorScreen/NewResourcePanel/NewResourceBaseForm.tsx [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+import WMInput from '../../../common/WMInput';
+import WMDropdown, { IWMDropdownOption } from '../../../common/WMDropdown';
+import WMButton from '../../../common/WMButton';
+
+import { ResourceOpenType, IResourceBaseData } from './interface';
+import { resourceOpenOptions, sizeUnitOptions } from './utils';
+
+import classes from './style.module.scss';
+
+export interface INewResourceForm {
```
Comment (nunibaranes) [inline src/components/Screen/CourseEditorScreen/NewResourcePanel/NewResourceBaseForm.tsx:20]: change to 'INewResourceBaseForm' and move it to interface.ts

File: src/components/Screen/CourseEditorScreen/NewResourcePanel/NewResourceBaseForm.tsx [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+  const [resourceData, setResourceData] = useState<IResourceBaseData>(initialNewResource);
+  const { title, url, openTarget, lightbox } = resourceData;
+
+  const resourceDataChange = (updated: Partial<IResourceBaseData>) =>
+    setResourceData((prev: IResourceBaseData) => ({
+      ...prev,
+      ...updated,
+    }));
+
+  const onUnitSize = (event: ChangeEvent<HTMLInputElement>, type: 'width' | 'height') => {
```
Comment (nunibaranes) [inline src/components/Screen/CourseEditorScreen/NewResourcePanel/NewResourceBaseForm.tsx:39]: onUnitSizechange

File: src/components/Screen/CourseEditorScreen/NewResourcePanel/NewResourceBaseForm.tsx [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+}
+
+export default function NewResourceBaseForm({
+  onDataChange,
+  initialNewResource,
+}: INewResourceForm): ReactElement {
+  const [resourceData, setResourceData] = useState<IResourceBaseData>(initialNewResource);
+  const { title, url, openTarget, lightbox } = resourceData;
+
+  const resourceDataChange = (updated: Partial<IResourceBaseData>) =>
```
Comment (nunibaranes) [inline src/components/Screen/CourseEditorScreen/NewResourcePanel/NewResourceBaseForm.tsx:33]: onResourceDataChange

File: src/components/Screen/CourseEditorScreen/NewResourcePanel/NewVideoForm.tsx [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+import FormGroup from '../../../common/FormGroup';
+import WMInput from '../../../common/WMInput';
+import WMSwitch from '../../../common/WMSwitch';
+
+import { IResourceBaseData, IResourceVideoData } from './interface';
+import NewResourceBaseForm from './NewResourceBaseForm';
+
+import classes from './style.module.scss';
+import { setVideoAutoplayParameter } from './utils';
```
Comment (nunibaranes) [inline src/components/Screen/CourseEditorScreen/NewResourcePanel/NewVideoForm.tsx:13]: Move before styles

File: src/components/Screen/CourseEditorScreen/NewResourcePanel/NewVideoForm.tsx [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+      ...resourceData,
+      ...updated,
+    });
+
+  useEffect(() => {
+    if (!_isEqual(initialNewResource, resourceData)) {
+      onDataChange(resourceData);
+    }
+  }, [resourceData, initialNewResource, onDataChange]);
+
```
Comment (nunibaranes) [inline src/components/Screen/CourseEditorScreen/NewResourcePanel/NewVideoForm.tsx:39]: Check the option to combine the form items to one form, according to the data (e.g: if autoplay property exist or videoParams show the proper fields)

File: src/components/Screen/CourseEditorScreen/NewResourcePanel/interface.ts [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+import { IWMDropdownOption } from '../../../common/WMDropdown';
+
+export type NewResourceType = CourseItemType.Article | CourseItemType.Video;
+
+export enum ResourceOpenType {
+  NewTab = 'new tab',
+  Lightbox = 'lightbox',
+}
+
+export enum SizeUnit {
```
Comment (nunibaranes) [inline src/components/Screen/CourseEditorScreen/NewResourcePanel/interface.ts:11]: Change to unit

File: src/components/Screen/CourseEditorScreen/NewResourcePanel/style.module.scss [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+
+  .actions {
+    margin: 25px auto;
+    display: flex;
+    justify-content: flex-end;
+    align-items: center;
+
+    :global(.ant-btn) {
+      &:last-child {
+        margin-left: 20px;
```
Comment (nunibaranes) [inline src/components/Screen/CourseEditorScreen/NewResourcePanel/style.module.scss:36]: Set default margin: 25px;
and for the last-child {margin-right: 0;}

File: src/components/Screen/CourseEditorScreen/NewResourcePanel/utils.ts [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+};
+
+export const initialNewVideoData: IResourceVideoData = {
+  ...initialNewResourceBaseData,
+  title: 'Untitled video',
+  autoplay: true,
+  videoPlayerParameters: 'autoplay="1"',
+};
+
+export const setVideoAutoplayParameter = ({
```
Comment (nunibaranes) [inline src/components/Screen/CourseEditorScreen/NewResourcePanel/utils.ts:38]: change name to replace or something

File: src/components/Screen/CourseEditorScreen/NewResourcePanel/utils.ts [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+  parameters,
+  autoplay,
+}: {
+  parameters: string;
+  autoplay: boolean;
+}): string => {
+  const strToReplace = autoplay ? 'autoplay="0"' : 'autoplay="1"';
```
Comment (nunibaranes) [inline src/components/Screen/CourseEditorScreen/NewResourcePanel/utils.ts:45]: Check if the parameters contain `autoplay` if not, add `autoplay="${checked}"`. else replace

File: src/components/Screen/CourseEditorScreen/ResourcesList.tsx [hunk 10]
```diff
@@ -15,7 +15,10 @@ import { SearchFilter } from '../../common/filters';
 
 import ResourceItemsList from './ResourceItemList';
 import ResourcesListEmptyState from './ResourcesListEmptyState';
+import ResourcesActionMenu from './ResourcesActionMenu';
+
 import classes from './style.module.scss';
+import NewResourcePanel, { NewResourceType } from './NewResourcePanel';
```
Comment (nunibaranes) [inline src/components/Screen/CourseEditorScreen/ResourcesList.tsx:21]: move before styles

File: src/components/Screen/CourseEditorScreen/style.module.scss [hunk 11]
```diff
@@ -65,9 +65,18 @@
     width: calc(var(--resources-list-width) - var(--resources-list-margin-size));
     margin-right: var(--resources-list-margin-size);
     background-color: var(--secondary200);
+    overflow-y: hidden;
```
Comment (nunibaranes) [inline src/components/Screen/CourseEditorScreen/style.module.scss:68]: Cleanup after refactor (new design)

File: src/components/Screen/Playground/components/playground.module.scss [hunk 12]
```diff
@@ -1,15 +1,51 @@
...
     }
   }
 }
 
 .cards-wrapper {
+  flex-grow: 1;
   display: flex;
   font-family: var(--font-primary);
   color: var(--neutral900);
+  height: calc(
```
Comment (nunibaranes) [inline src/components/Screen/Playground/components/playground.module.scss:14]: Removeeeeee

File: src/components/Screen/Playground/style.module.scss [hunk 13]
```diff
@@ -0,0 +1,3 @@
+.playground {
```
Comment (nunibaranes) [inline src/components/Screen/Playground/style.module.scss:1]: removeeeee

File: src/components/common/DetailsPanel/style.module.scss [hunk 14]
```diff
@@ -64,3 +64,13 @@
   flex-grow: 1;
   overflow: auto;
 }
+
+@keyframes fadeInDown {
```
Comment (nunibaranes) [inline src/components/common/DetailsPanel/style.module.scss:68]: Removeeeee

File: src/components/common/WMRadio/interface.ts [hunk 15]
```diff
@@ -6,6 +6,7 @@ export interface IRadioButton {
   label: ReactNode;
   disabled?: boolean;
   helpText?: string;
+  content?: ReactNode;
```
Comment (nunibaranes) [inline src/components/common/WMRadio/interface.ts:9]: Removeeee

File: src/components/common/dialogs/PublishToEnvironmentDialog/index.tsx [hunk 16]
```diff
@@ -42,10 +42,10 @@ export default function PublishToEnvironmentDialog({
         </div>
       }
       confirmClass={classes['confirmation-button']}
-      confirmLabel={`Publish to ${environment.value}`}
-      onConfirm={() =>
-        onConfirm({ envId: environment.id, envName: environment.label ?? environment.value })
-      }
+      confirmLabel={environment && `Publish to ${environment.value}`}
+      onConfirm={() => {
```
Comment (nunibaranes) [inline src/components/common/dialogs/PublishToEnvironmentDialog/index.tsx:46]: remove curly braces

File: src/utils/index.ts [hunk 17]
```diff
@@ -8,3 +8,16 @@ export const pluralizer = (word: string, count: number): string =>
...
+  return (!isNaN(parseInt(value)) && reg.test(value)) || value === '' || value === '-';
+};
+
+export const getValidRangeNumber = (value: string, start = 0, end = 100): number =>
+  value === '' || value === '-' ? start : parseInt(value) > end ? end : parseInt(value);
+
+export const fieldErrorMessage = (
+  fieldValue: string,
+  error = 'This field cannot be empty',
+): string | undefined => (fieldValue === '' ? error : undefined);
```
Comment (nunibaranes) [inline src/utils/index.ts:23]: move form related helpers to a separate file (form.ts)

File: src/components/Screen/CourseEditorScreen/CourseOutlineDetailsPanel.tsx [hunk 18]
```diff
@@ -24,6 +26,22 @@ export const TaskItemIconType = {
...
   const [{ course, isDetailsPanelOpen, activeDetailsItem }, dispatch] = useCourseEditorContext();
+  const [newResourceData, setNewResourceData] = useState<IResourceBaseData | IResourceVideoData>();
+  const [hasValidationError, setHasValidationError] = useState<boolean>(false);
+
+  const resourceIcon = activeDetailsItem && <Icon type={activeDetailsItem.type} />;
+
+  const isValidData = (data: IResourceBaseData | IResourceVideoData) => {
+    const isEmptyStr = (val?: string) => val === '';
+
+    return isEmptyStr(data.title) || isEmptyStr(data.url);
```
Comment (dvira) [inline src/components/Screen/CourseEditorScreen/CourseOutlineDetailsPanel.tsx:37]: Using the string's length property should do the same no?
e.g.: `!data.title.length`.

File: src/components/Screen/CourseEditorScreen/NewResourcePanel/utils.ts [hunk 19]
```diff
@@ -0,0 +1,56 @@
...
+};
+
+export const initialNewVideoData: IResourceVideoData = {
+  ...initialNewResourceBaseData,
+  title: 'Untitled video',
+  autoplay: true,
+  videoPlayerParameters: '',
+};
+
+// TODO: if the following variables / methods not unnecessary remove them
```
Comment (dvira) [inline src/components/Screen/CourseEditorScreen/NewResourcePanel/utils.ts:40]: FYI, "not unnecessary" means "necessary".
I think you meant "are unnecessary", right?

File: src/components/Screen/CourseEditorScreen/NewResourcePanel/utils.ts [hunk 20]
```diff
@@ -0,0 +1,56 @@
...
+
+export const autoplayActiveStr = 'autoplay="1"';
+
+/**
+ * video parameters should effects according to autoplay value
```
Comment (dvira) [inline src/components/Screen/CourseEditorScreen/NewResourcePanel/utils.ts:45]: This sentence makes little sense, did you mean to say: 
"Video parameters would be affected by the autoplay value"?

**Review by dvira — APPROVED** [review]
Looks good! 
Had a few minor comments..

**Pull request merged**